Mathlib/FieldTheory/Adjoin.lean

```diff
@@ -729,6 +730,30 @@ instance finiteDimensional_sup [FiniteDimensional K E1] [FiniteDimensional K E2]
     rwa [this] at h
   rw [Algebra.TensorProduct.productMap_range, E1.range_val, E2.range_val, sup_toSubalgebra_of_left]
 
+/-- If `E1` and `E2` are intermediate fields, and at least one them are algebraic, then the rank of
+`E1 ⊔ E2` is less than or equal to the product of that of `E1` and `E2`. Note that this result is
+also true without algebraic assumption, but the proof becomes very complicated. -/
+theorem rank_sup_le_of_isAlgebraic
+    (halg : Algebra.IsAlgebraic K E1 ∨ Algebra.IsAlgebraic K E2) :
+    Module.rank K ↥(E1 ⊔ E2) ≤ Module.rank K E1 * Module.rank K E2 := by
```

Comment:
LGTM! Let's merge this now; when you're able to remove the algebraic condition, it would be reasonable to import AlgebraicIndependent, which adds 3 imports, plus 1 for RelRank.

Reply:
> it would be reasonable to import AlgebraicIndependent, which adds 3 imports, plus 1 for RelRank

Unfortunately, that's not possible, since `AlgebraicIndependent` (`aevalEquivField` in my working PR) and `RelRank` imports `Mathlib.FieldTheory.Adjoin` 😂